### ISSUE TITLE
Upgrade NiFi version to 1.15.0

### ISSUE THREAD

**Issue by david-streamlio** [david-streamlio/pulsar-nifi-bundle#32]
Upgrade the library version and implement the new  validation mechanism.

**Comment by ramkumarkb** [comment]
Since most installations of NiFi would need to upgrade to v1.15.2 (for [the log4j remediation](https://cwiki.apache.org/confluence/display/NIFI/Release+Notes#ReleaseNotes-Version1.15.2)) - this work would be valuable indeed. Also, if one needs to give a PR for 1.15.2 - what would be the broad guidelines for such piece of work?

Thank you!

**Comment by david-streamlio** [comment]
@ramkumarkb I agree 100%. I had originally planned on incorporating the addition of the new validation methods added in the 1.15 release to this PR, but given the security issue, I think it is best to just focus on upgrading the NiFi version to 1.15.2.
